packages/vue-components/src/PageNavButton.vue

```diff
@@ -0,0 +1,98 @@
+<template>
+  <div v-show="hasPageNav">
+    <span
+      id="toggle-page-nav-button"
+      :class="['glyphicon', { 'nav-menu-close-icon': show }]"
+      @click="togglePageNav"
+    ></span>
+    <div ref="pageNavMenu" :class="['page-nav-menu', { 'nav-menu-open': show }]"></div>
+  </div>
+</template>
+
+<script>
+import $ from './utils/NodeList';
+
+export default {
+  data() {
+    return {
+      hasPageNav: false,
+      show: false,
+    };
+  },
+  inject: ['initPageNav'],
+  methods: {
+    togglePageNav() {
+      this.show = !this.show;
+      this.toggleSiteNavButton();
+    },
+    toggleSiteNavButton() {
+      const siteNavButton = document.getElementById('toggle-site-nav-button');
```

Comment:
would `:class` work? Similar to what you did for `nav-menu-open` above. (also for the page nav variant)

Reply:
Hmm I'm not sure if `:class` would work for this because this is querying `page-nav-button` from `site-nav` when it is opened. They should be considered sibling elements so not sure how to pass the data to each other unless we pass it to the parent and then back to the child. Will just using `getElementById` be better because the buttons `id` are fixed and will not be changed by the user 🤔?

Reply:
ahh, my bad. I didn't notice that this was for the other component at all. 🙈 


> They should be considered sibling elements so not sure how to pass the data to each other unless we pass it to the parent and then back to the child.

I agree this is a little messy (and again reintroduces coupling with the navbar).

my concern here is with trying to limit dom effects that can be limited to vue to be limited to vue, to minimise side effects outside of vue. (as opposed to interfacing with `<ul>` / `<li>` in dropdowns - no choice there)

but introducing global state management for this seems overkill as well.. hmm.......

Reply:
how about just a event subscribe / consume implementation?

```
// pagenavbutton / sitenavbutton
subscribe('closeoverlays', () => this.show = false)
publish('closeoverlays')

// simple event implementation
window.subscribers = [ ... ]
......... or just this https://developer.mozilla.org/en-US/docs/Web/Guide/Events/Creating_and_triggering_events
```

this way the dom manipulation stays within the subscriber component (page / site button); won't need to tie the implementation to the classes and ids as well

Reply:
Thanks for the suggestion 😄